typing/parmatch.ml

```diff
@@ -1132,10 +1115,17 @@ let rec do_match pss qs = match qs with
   | {pat_desc = Tpat_or (q1,q2,_)} ->
       do_match pss (q1::qs) || do_match pss (q2::qs)
   | {pat_desc = Tpat_any} ->
-      do_match (do_filter_var pss) qs
+      let rec remove_first_column = function
+        | (_::ps)::rem -> ps::remove_first_column rem
+        | _ -> []
+      in
```

Comment:
Is this the same thing as `List.map List.tl`?

Reply:
Not exactly: what about matrices which contains some rows but no columns?
I don't know off-hand whether we can have some in that places and the current implementation was meant to handle them, or if the code was just written this way because it was written this way.

Not sure it's worth changing.